cmd/installer/cli/join2.go

```diff
@@ -0,0 +1,584 @@
+// Applies first the EmbeddedOverrides and then the EndUserOverrides.
+func applyJoinConfigurationOverrides(jcmd *kotsadm.JoinCommandResponse) error {
+	patch, err := jcmd.EmbeddedOverrides()
+	if err != nil {
+		return fmt.Errorf("unable to get embedded overrides: %w", err)
+	}
+	if len(patch) > 0 {
+		if data, err := yaml.Marshal(patch); err != nil {
+			return fmt.Errorf("unable to marshal embedded overrides: %w", err)
+		} else if err := patchK0sConfig(
+			runtimeconfig.PathToK0sConfig(), string(data),
+		); err != nil {
+			return fmt.Errorf("unable to patch config with embedded data: %w", err)
+		}
+	}
+	if patch, err = jcmd.EndUserOverrides(); err != nil {
+		return fmt.Errorf("unable to get embedded overrides: %w", err)
+	} else if len(patch) == 0 {
+		return nil
+	}
+	if data, err := yaml.Marshal(patch); err != nil {
+		return fmt.Errorf("unable to marshal embedded overrides: %w", err)
+	} else if err := patchK0sConfig(
+		runtimeconfig.PathToK0sConfig(), string(data),
+	); err != nil {
+		return fmt.Errorf("unable to patch config with embedded data: %w", err)
+	}
+	return nil
+}
+
+// patchK0sConfig patches the created k0s config with the unsupported overrides passed in.
+func patchK0sConfig(path string, patch string) error {
+	if len(patch) == 0 {
+		return nil
+	}
+	finalcfg := k0sconfig.ClusterConfig{
+		ObjectMeta: metav1.ObjectMeta{Name: runtimeconfig.BinaryName()},
+	}
+	if _, err := os.Stat(path); err == nil {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("unable to read node config: %w", err)
+		}
+		finalcfg = k0sconfig.ClusterConfig{}
+		if err := k8syaml.Unmarshal(data, &finalcfg); err != nil {
+			return fmt.Errorf("unable to unmarshal node config: %w", err)
+		}
+	}
+	result, err := config.PatchK0sConfig(finalcfg.DeepCopy(), patch)
+	if err != nil {
+		return fmt.Errorf("unable to patch node config: %w", err)
+	}
+	if result.Spec.API != nil {
+		if finalcfg.Spec == nil {
+			finalcfg.Spec = &k0sconfig.ClusterSpec{}
+		}
+		finalcfg.Spec.API = result.Spec.API
+	}
+	if result.Spec.Storage != nil {
+		if finalcfg.Spec == nil {
+			finalcfg.Spec = &k0sconfig.ClusterSpec{}
+		}
+		finalcfg.Spec.Storage = result.Spec.Storage
+	}
+	// This is necessary to install the previous version of k0s in e2e tests
+	// TODO: remove this once the previous version is > 1.29
+	unstructured, err := helpers.K0sClusterConfigTo129Compat(&finalcfg)
+	if err != nil {
+		return fmt.Errorf("unable to convert cluster config to 1.29 compat: %w", err)
+	}
+	data, err := k8syaml.Marshal(unstructured)
+	if err != nil {
+		return fmt.Errorf("unable to marshal node config: %w", err)
+	}
+	if err := os.WriteFile(path, data, 0600); err != nil {
+		return fmt.Errorf("unable to write node config file: %w", err)
+	}
+	return nil
+}
+
+// runK0sInstallCommand runs the k0s install command as provided by the kots
+// adm api.
+func runK0sInstallCommand(cmd *cobra.Command, fullcmd string) error {
```

Comment:
don't pass cmd here, let's pass the specific flags instead (appears to be just networkInterface)?